help/develop/domains/chem/cheminformatics.md

```diff
@@ -31,59 +31,36 @@ descriptors(table, column, descriptors);
 
 The `settings` object allows passing the following parameters:
 
-* `substructLibrary`: a boolean indicating whether to use the "naive" graph-based search method,
-  using [`get_substruct_match`](https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html) method of
-  RDKit (`substructLibrary: false`
-  , [code sample 1](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-simple)
-  , [code sample 2](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-library))
-  , or a [`substructLibrary`](https://rdkit.blogspot.com/2018/02/introducing-substructlibrary.html)
-  from RDKit (`substructLibrary: true`
-  , [code sample](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-library)). The default value
-  is `true`.
-
-For the chosen `substructLibrary` option, the first call to the function with a `column` previously unseen by
-the `searchSubstructure` function shall build a library. The library is used for further searches in the same given
-column while the `pattern` is varying.
-
-The function's user doesn't need to manage the built library, as it is done by the function automatically. As long as
-the function detects it has received a column different to the one previously seen, it shall rebuild the library for
-this newly seen column before continuing with searches.
-
-It is possible to call `searchSubstructure` solely for initializing the library, without performing an actual search. In
-order to do this, `pattern` should be passed with a value of either `null` or an empty string `'''`.
-
-Sometimes it happens that the molecule strings aren't supported by RDKit-JS. These inputs are skipped when indexing, but
-an error log entry is added for each such entry to the browser's console together with the actual molecule string. For
-such unsupported molecule, the corresponding bit in the bitset remains as 0. Thus, the bitset is *always* of the
+* `molBlockFailover`: smarts which used as a substructure if `pattern` is invalid
+
+To make search maximum efficient substructure search function maintains a cache of
+pre-computed Pattern fingerprints (a topological fingerprint optimized for substructure screening) and specific RDKit *Mol objects*. Fingerprints are used for preliminary filtration while *Mol objects* are used for graph-based search method [`get_substruct_match`](https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html). Fingerprints are pre-calculated for the whole column while *Mol objects* are cached only for defined number of first molecules. One the one hand it allows to decrease memory consumption and on the other hand speeds up the search. Once the function `searchSubstructure` meets the `column` previously unmet, it builds the cache of fingerprints and *Mol objects* for the molecules in `column`. Later it is used for substructure search as long as the function is invoked for the same `column`.
```

Comment:
No need to write the description of Pattern fingerprints? I suggest just to add a link.

Reply:
What for do we need `One the one hand` here? And `on the other hand`?
Just:
`It allows to decrease memory consumpution and speed up the search`. 
Or even:
`This approach decreases memory consumption and speeds up the search`.

Reply:
`meets the column previously unmet` is too hard to understand. 
I suggest to simplify this sentence:
`When the substructure search function encounters a new column, it creates a cache of fingerprints and Mol objects for the molecules in it`.

Reply:
Lets simplify the sentence starting with Later. We should make it as simple as possible.
Something like this:
`Substructure search uses this cache (it) when the function is called for the same column.`

Reply:
Done